gpx export feature

MichaelsJP opened the pull request:
gpx export feature integrated

Commit:
gpx export feature

rabidllama commented on openrouteservice/src/main/java/heigit/ors/util/gpxUtil/PtType.java:
```diff
@@ -0,0 +1,146 @@
+
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.annotation.*;
```
We should avoid importing all items from the package to keep things clean

TimMcCauley replied:
Yes please, let's think lean!

rabidllama commented on openrouteservice/src/main/java/heigit/ors/util/gpxUtil/LinkType.java:
```diff
@@ -0,0 +1,116 @@
+
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.annotation.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama commented on openrouteservice/src/main/java/heigit/ors/util/gpxUtil/Gpx.java:
```diff
@@ -0,0 +1,251 @@
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.annotation.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama commented on openrouteservice/src/main/java/heigit/ors/util/gpxUtil/CopyrightType.java:
```diff
@@ -0,0 +1,120 @@
+
+package heigit.ors.util.gpxUtil;
+
+import javax.xml.bind.annotation.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama commented on openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/json/JsonRoutingResponseWriter.java:
```diff
@@ -20,33 +20,21 @@
-import heigit.ors.routing.RoutingProfileType;
-import heigit.ors.routing.RoutingRequest;
-import heigit.ors.routing.WeightingMethod;
+import heigit.ors.routing.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama commented on openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java:
```diff
@@ -0,0 +1,166 @@
+import heigit.ors.services.routing.RoutingServiceSettings;
+import heigit.ors.util.AppInfo;
+import heigit.ors.util.GeomUtility;
+import heigit.ors.util.gpxUtil.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama commented on openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/gpx/GpxRoutingResponseWriter.java:
```diff
@@ -0,0 +1,166 @@
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.Point;
+import heigit.ors.config.AppConfig;
+import heigit.ors.routing.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama commented on openrouteservice/src/main/java/heigit/ors/services/routing/requestprocessors/RoutingRequestParser.java:
```diff
@@ -18,35 +18,26 @@
-import heigit.ors.routing.RoutingRequest;
-import heigit.ors.routing.WayPointBearing;
-import heigit.ors.routing.WeightingMethod;
+import heigit.ors.routing.*;
```
We should avoid importing all items from the package to keep things clean

rabidllama reviewed (changes requested):
Mostly looks good, main thing I would say is that we should try to not use the .* for imports. It doesn't affect performance, but it can lead to issues down the line if there are clashes between packages.

Also, we should think about adding the unit testing to it.

TimMcCauley commented:
@MichaelsJP let's add the unit tests, it's a good start.

Commit:
corrected all &lt with <
replaced all .*; imports with multiline imports
added unit test for XMLBuilderTest.java

rabidllama committed:
Update BoundsType.java

rabidllama committed:
Update CopyrightType.java

rabidllama committed:
Update EmailType.java

rabidllama committed:
Update BoundsType.java

rabidllama committed:
Update CopyrightType.java

rabidllama committed:
Update EmailType.java

rabidllama committed:
Update LinkType.java

rabidllama committed:
Update MetadataType.java

rabidllama committed:
Update PersonType.java

rabidllama committed:
Update PtType.java

rabidllama committed:
Update PtsegType.java

rabidllama committed:
Update RteType.java

rabidllama committed:
Update TrkType.java

rabidllama committed:
Update TrksegType.java

rabidllama committed:
Update WptType.java
